app/src/main/java/com/codepath/michfeng/songswiper/fragments/SwipeFragment.java

```diff
@@ -70,11 +79,30 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         tv = (TextView) view.findViewById(R.id.textView);
 
         recommendationService = new RecommendationService(getApplicationContext());
-        recommendationService.get(() -> {
-            this.recommendations = recommendationService.getRecommendations();
-        });
 
-        // At this point, recommendations is still null. Possibly an issue with scope.
+        String accessToken = getArguments().getString("accessToken");
+        Log.i(TAG,"access token: "+accessToken);
+        /*recommendationService.get(() -> {
+            recommendations = recommendationService.getRecommendations();
+            Log.i("SwipeFragment","recommended tracks: "+recommendations.toString());
+        });*/
+
+
+        SpotifyApi spotifyApi = new SpotifyApi(accessToken);
+
+        List<TrackFull> topTrackFull = spotifyApi.getTopTracks(new HashMap<>()).getItems();
+        List<ArtistFull> topArtistFull = spotifyApi.getTopArtists(new HashMap<>()).getItems();
+        List<String> topGenres = new ArrayList<>();
+        List<String> topTracks = new ArrayList<>();
+        List<String> topArtists = new ArrayList<>();
+
+        for (TrackFull t : topTrackFull) topTracks.add(t.getId());
+        for (ArtistFull a : topArtistFull) {
+            topGenres.addAll(a.getGenres());
```

Comment:
how are you keeping track of duplicate genres? do you re-use topArtistFull to find out the genres of each artist?